pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1750,3 +1770,86 @@ func (r *ReconcileClusterDeployment) deleteStaleProvisions(provs []*hivev1.Clust
 		}
 	}
 }
+
+// getAllSyncSetInstances return all syncset instances for a cluster deployment
+func (r *ReconcileClusterDeployment) getAllSyncSetInstances(cd *hivev1.ClusterDeployment) ([]*hivev1.SyncSetInstance, error) {
+
+	list := &hivev1.SyncSetInstanceList{}
+	err := r.Client.List(context.TODO(), list, client.InNamespace(cd.Namespace))
+	if err != nil {
+		return nil, err
+	}
+
+	syncSetInstances := []*hivev1.SyncSetInstance{}
+	for i, syncSetInstance := range list.Items {
+		fmt.Println("syncsetinstance name: ", syncSetInstance.Name)
+		if syncSetInstance.Spec.ClusterDeployment.Name == cd.Name {
```

Comment:
We should add the `constants.ClusterDeploymentNameLabel` label to the syncsetinstance so that the `List` returns only the relevant resources. Then, this name comparison would not be needed. It is good practice to have that label on the syncsetinstance anyway.

Reply:
+1, sounds good to me. IMO I should fix this post this PR

Reply:
Also we should add the label to the syncset objects as well, isn't? @abutcher WDYT?

Reply:
I don't know that the syncset resources should be labeled unless we are going to use that to associate the syncset with a clusterdeployment instead of using the reference in the spec of the syncset. The syncset resource is different that the syncsetinstance since one is created by the user and the other is created by the controller.

Reply:
Good point